nexus_transporter/src/transporters/rmf_transporter.cpp

```diff
@@ -251,18 +393,42 @@ class RmfTransporter : public Transporter
       _itinerary_expiration_seconds
     );
 
+    // The internal node is used to run any interactions with RMF without
+    // being blocked by the transporter node's actions and services
+    _internal_node = rclcpp::Node::make_shared("rmf_transporter_internal_node");
+
+    _spin_thread = std::make_shared<std::thread>([&]()
+      {
+        while (rclcpp::ok())
+        {
+          rclcpp::spin_some(_internal_node);
+          std::this_thread::sleep_for(std::chrono::milliseconds{100});
+        }

```

Comment:
I think we can avoid this `spin_some` pattern with just `spin` since this is running in a separate thread. `spin()` should exit when a Ctrl+C signal is registered.

I kinda want to try the `[EventsExecutor](https://github.com/ros2/rclcpp/blob/b7e4aad091c6b334b9ad17a8f83007ea9901cc3b/rclcpp/include/rclcpp/experimental/executors/events_executor/events_executor.hpp#L73)` here for better performance. It's still "experimental" but I think it would be perfect for this sort of internal node. Shall we give this a shot?
```
 rclcpp::experimental::executors::EventsExecutor executor;
 executor.add_node(node);
 executor.spin();
```

Reply:
great idea, swapped it over in https://github.com/osrf/nexus/pull/110/commits/4415f9edee6985d02b33e7d0757eb99b583ee5cb